lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -210,17 +213,31 @@ def test_no_ccx_created(self):
     def test_create_ccx(self):
         """
         Create CCX. Follow redirect to coach dashboard, confirm we see
-        the coach dashboard for the new CCX.
+        the coach dashboard for the new CCX. Also, confirm that the coach
+        has staff access to it.
         """
+
         self.make_coach()
         url = reverse(
             'create_ccx',
-            kwargs={'course_id': self.course.id.to_deprecated_string()})
+            kwargs={'course_id': unicode(self.course.id)})
+
         response = self.client.post(url, {'name': 'New CCX'})
         self.assertEqual(response.status_code, 302)
         url = response.get('location')  # pylint: disable=no-member
         response = self.client.get(url)
         self.assertEqual(response.status_code, 200)
+
+        # To locate the ccx_key, first get the path from the URL
+        path = urlparse.urlparse(url).path
+        
+        # Then get the second to last segment of the path
+        head, tail = os.path.split(path)
+        head, ccx_key = os.path.split(head)
```

Comment:
The reason I suggested you use `resolve()` when parsing URLs is that it's going to be more robust than manually parsing it yourself, especially if we decide to move URLs around. As a somewhat silly example, say we try to run this on Windows, where the path separator is a `\` instead of a `/`. If you're parsing a URL, you might as well use exactly Django's logic for doing so.

```python
>>> from django.core.urlresolvers import resolve
>>> resolver = resolve("/courses/ccx-v1:DavidsonNext+Cal_APccx_Edge+3T2015+ccx@114/ccx_coach")
>>> resolver.kwargs['course_id']
'ccx-v1:DavidsonNext+Cal_APccx_Edge+3T2015+ccx@114'
```